src/result.js

```diff
@@ -150,15 +150,20 @@ class Result {
     }
   }
 
-  _destroyAppInstance() {
-    if (this.instance && !this._instanceDestroyed) {
-      this._instanceDestroyed = true;
-      this.instance.destroy();
-
+  _destroy() {
+    if (this.isDestroyed) {
       return true;
     }
 
-    return false;
+    this.isDestroyed = true;
+
+    if (this.applicationInstanceInstance !== undefined) {
+      this.applicationInstanceInstance.destroy();
+    }
+
+    if (this.applicationInstance !== undefined) {
+      this.applicationInstance.destroy();
+    }
```

Comment:
@kratiahuja - This removes the conditionalness (if that is a word), we _always_ call destroy and what we used to use the return value for isn't needed any more. Am I missing something?

Reply:
I think the place where we destroy the application with a timer (`destroyAppInstanceInMs`): `if (result._destroy()) {`, that would never run because we aren't return a `true` value.

Reply:
Put a new comment to point to the actual code which will never run.